src/libtexture/texturesys.cpp

```diff
@@ -1264,16 +1264,16 @@ TextureSystemImpl::texture_lookup_trilinear_mipmap (TextureFile &texturefile,
         if (! levelweight[level])  // No contribution from this level, skip it
             continue;
         float4 r, drds, drdt;
-        weight[0] = levelweight[level];
         ok &= (this->*sampler) (1, sval, tval, miplevel[level],
                                 texturefile, thread_info, options,
                                 nchannels_result, actualchannels, weight,
                                 &r, dresultds ? &drds : NULL, dresultds ? &drdt : NULL);
         ++npointson;
-        *(simd::float4 *)(result) += r;
+        float lw = levelweight[level];
```

Comment:
I think this will be slightly faster as

    float4 lw = levelweight[level];

mixing scalar float and vector float is more expensive than vector/vector. Since lw might get used 3 times (for the deriv case), it's better to store it as a float4 once.

I can make this fix on my end, just letting you know.